crates/outbound-http/src/lib.rs

```diff
@@ -21,6 +21,8 @@ pub const ALLOWED_HTTP_HOSTS_KEY: MetadataKey<Vec<String>> = MetadataKey::new("a
 pub struct OutboundHttp {
     /// List of hosts guest modules are allowed to make requests to.
     pub allowed_hosts: AllowedHttpHosts,
+    /// origin is the host of the component that created this outbound HTTP request.
```

Comment:
This struct isn't associated with a specific HTTP request, so we could make this more explicit:

```suggestion
    /// During an incoming HTTP request, origin is set to the host of that incoming HTTP request.
    /// This is used to automatically allow outbound requests to the same host.
```

Reply:
that's really great wording thank you